ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -143,14 +150,22 @@ private async Task<ImportResult> ValidateAndImportNotificationGroupAsync(Perform
             }
 
             _logger.LogSuccess(context, requestId, $"Importing {notifications.Count()} valid notifications");
-            var savedNotifications = await _notificationImportRepository.AddLinkedNotificationsAsync(notifications);
-            await _migrationRepository.MarkNotificiationsAsImportedAsync(savedNotifications);
+            try
+            {
+                var savedNotifications = await _notificationImportRepository.AddLinkedNotificationsAsync(notifications);
+                await _migrationRepository.MarkNotificiationsAsImportedAsync(savedNotifications);
 
 
-            var newIdsString = string.Join(" ,", savedNotifications.Select(x => x.NotificationId));
-            _logger.LogSuccess(context, requestId, $"Imported notifications have following Ids: {newIdsString}");
+                var newIdsString = string.Join(" ,", savedNotifications.Select(x => x.NotificationId));
+                _logger.LogSuccess(context, requestId, $"Imported notifications have following Ids: {newIdsString}");
 
-            _logger.LogInformation(context, requestId, $"Finished importing notification for {patientName}");
+                _logger.LogInformation(context, requestId, $"Finished importing notification for {patientName}");
+            }
+            catch (Exception e)
+            {
+                _logger.LogFailure(context, requestId, message: $"Failed to save the imported notifications for {patientName}", e);

```

Comment:
The failure can also be marking the notification. In that cases the notification will be saved but not marked as imported. But failure message will show it as not saved

Reply:
ah, that's a good shout - i'll make the message better, though it's probably still a fine for it to be flagged this way, as it'll draw our attention to it nevertheless